src/main/scala/mimir/algebra/gprom/OperatorTranslation.scala

```diff
@@ -1867,7 +1867,7 @@ object OperatorTranslation {
     var listTail : GProMListCell.ByReference = null
     var i = 0;
     
-    val schema = db.bestGuessSchema(oper)
+    val schema = db.schemaOf(oper)
     for(schemaTup : (String, Type) <- schema.reverse){
       val attrName = schemaTup._1
```

Comment:
Here (and possibly in a few places above), couldn't you use Operator's `columnNames` method?  In any case where you don't need the type information, columnNames is faster, cheaper, and doesn't require `db`.